graphql/src/graphql_schema.ml

```diff
@@ -329,14 +338,36 @@ module Make(Io : IO) = struct
   and ('ctx, 'a) abstract_value =
     AbstractValue : ('ctx, 'src option) typ * 'src -> ('ctx, 'a) abstract_value
 
+  type 'ctx subscription_field =
+    SubscriptionField : {
+      name       : string;
+      doc        : string option;
+      deprecated : deprecated;
+      typ        : ('ctx, 'out) typ;
+      args       : (('out stream, string) result, 'args) Arg.arg_list;
```

Comment:
I would make this `('out stream, string) result io` to allow async operations to create the subscription. This is required e.g. for `ocaml-redis`.

Reply:
I'm in favor of adding a `subscription_io_field` like we have for regular fields. I think some programs will want to have streams of values as currently implemented, while others like OCaml-redis would need `'out stream io`s.

I'll add an implementation in my next commit for you to see what you think